packages/nouns-webapp/src/pages/Oracle.tsx

```diff
@@ -0,0 +1,307 @@
+import { Button } from '@/components/ui/button';
+import { Progress } from '@/components/ui/progress';
+import {
+  readNounsAuctionHouseAuction,
+  useWriteNounsAuctionHouseSettleCurrentAndCreateNewAuction,
+} from '@/contracts';
+import { useAppDispatch, useAppSelector } from '@/hooks';
+import { setStateBackgroundColor } from '@/state/slices/application';
+import { RootState } from '@/store';
+import { beige, grey } from '@/utils/nounBgColors';
+import { config } from '@/wagmi';
+
+enum AuctionState {
+  NotStarted,
+  Active,
+  OverNotSettled,
+  OverAndSettled,
+}
+
+type Auction = {
+  nounId: number;
+  endTime: Date;
+  startTime: Date;
+  amount: string;
+  settled: boolean;
+  bidder: Address;
+  state: AuctionState;
+};
+
+type Noun = {
+  nounId: number;
+  seed: ReturnType<typeof getNounSeedFromBlockHash>;
+  image: string;
+};
+
+const determineAuctionState = (
+  startTime: number,
+  settled: boolean,
+  currentTime: number,
+  endTime: number,
+): AuctionState => {
+  if (startTime === 0) return AuctionState.NotStarted;
+  if (settled) return AuctionState.OverAndSettled;
+  if (currentTime < endTime) return AuctionState.Active;
+  return AuctionState.OverNotSettled;
+};
+
+const processNounData = (nounId: number, blockHash: string) => {
+  const nounSeed = getNounSeedFromBlockHash(nounId, blockHash);
+  const { parts, background } = getNounData(nounSeed);
+  const image = `data:image/svg+xml;base64,${btoa(buildSVG(parts, imageData.palette, background))}`;
+  return {
+    nounId,
+    seed: nounSeed,
+    image,
+    backgroundColor: nounSeed.background === 0 ? grey : beige,
+  };
+};
+
+const createAuctionData = (
+  auctionData: {
+    nounId: bigint;
+    amount: bigint;
+    startTime: number;
+    endTime: number;
+    bidder: Address;
+    settled: boolean;
+  },
+  state: AuctionState,
+): Auction => ({
+  nounId: Number(auctionData.nounId),
+  endTime: dayjs.unix(auctionData.endTime).toDate(),
+  startTime: dayjs.unix(auctionData.startTime).toDate(),
+  amount: formatEther(auctionData.amount),
+  settled: auctionData.settled,
+  bidder: auctionData.bidder,
+  state,
+});
+
+export function OraclePage() {
+  const { t } = useLingui();
+  const dispatch = useAppDispatch();
+  const stateBgColor = useAppSelector((state: RootState) => state.application.stateBackgroundColor);
+
+  const [auction, setAuction] = useState<Auction | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [nextNoun, setNextNoun] = useState<Noun | null>(null);
+  const [currentBlockTimestamp, setCurrentBlockTimestamp] = useState<number>(0);
+  const prevNounIdRef = useRef<number | null>(null); // Add this to track the previous noun ID
+
+  const {
+    writeContract: settleAuction,
+    isPending: isSettlingAuction,
+    isError: didSettleFail,
+    error: settleAuctionError,
+  } = useWriteNounsAuctionHouseSettleCurrentAndCreateNewAuction();
+
+  useWatchBlocks({
+    syncConnectedChain: false,
```

Comment:
should probably set this up globally on `wagmi.ts`
(I thought it was already 🤔 )